### PR TITLE
Creating Payment Methods for Future Purchases

### LINKED ISSUE #100: FS: Payment Methods
Changes include the following:
- We'll need to update the Profile model too - attach the stripeCustomerId onto the Profile.
- Using [Setup Intents](https://stripe.com/docs/api/setup_intents) let's design our Payment Methods page for a customer.
**Back-end:**
- Let's add a new payment methods controller. 
- A route we will need is `POST /payment_methods/session` **Note: Due to simplicity I am going to allow the snake case URL**, this will create a **Setup Intent** checkout session with Stripe. Due to timing, we're going to focus on leveraging Stripe's UI for this.
- We'll also need some routes for successful returns from Stripe Session.
- Learn more about Setup Intents in this video - https://www.youtube.com/watch?v=S1WJuhGkY88
- **Investigate how we can attach the payment method to the customer - this shouldn't be too hard since Stripe should deal with it but I am not 100% sure on the process.**
**Front-end**:
- We'll need a new settings page for the payment methods. Let's ensure to render the customers payment methods, and a button to add a new payment method.

### PR DESCRIPTION
**Pull request by RetiFier**
### What this PR does (required):
- Can Save and Retrieve Payment Methods for Future Purchases from stripe

### Screenshots / Videos (front-end only):
image
image

### Any information needed to test this feature (required):
- In Setting Payments Methods can add New Payments Methods. 
- After Adding the Payments, the saved payments will show 
-  http://localhost:3001/payment_methods/session POST to create new payments methods 
- http://localhost:3001/payment_methods/session GET to get the saved payments methods

### Any issues with the current functionality (optional):
- This Branch is branch from stripe-connect branch
- The Stripe Connect Component will works after adding role base nav bar.

### CODE REVIEW

File: client/src/pages/Settings/Settings.tsx [hunk 1]
```diff
@@ -25,10 +27,18 @@ const settingsMenu = [
     to: '/profile/settings/availability',
     component: <SettingHeader header="Availability" />,
   },
+  // To use after implementing role base nav bar
+  // {
+  //   name: 'Payment methods',
+  //   to: '/profile/settings/payment-methods',
+  //   canView: [AccountType.PET_SITTER],
+  //   component: <StripeConnect header="Payment Methods" />,
+  // },
```
Comment (sundayezeilo) [inline client/src/pages/Settings/Settings.tsx]: Please move unused code away from here

File: server/controllers/paymentMethod.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+const stripe = require("stripe")(process.env.STRIPE_SECRET);
+
+// @route POST /payment_methods/session
+// @desc create payment for future
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/paymentMethod.js:7]: Good documentation

File: server/controllers/paymentMethod.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+
+
+// @route GET /payment_methods/session
+// @desc GET saved payment
+// @access Private
+exports.listPaymentSession = asyncHandler(async (req, res, next) => {
+  const id = req.user.id;
+  const getProfile = await Profile.findOne({ userId: id });
+  console.log(getProfile?.stripeCustomerId);
```
Comment (sundayezeilo) [inline server/controllers/paymentMethod.js]: Please watch for unnecessary logs to the console. Going forward, we should be cleaning up our logs done during development. Thanks!

File: server/controllers/stripe.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          success: {
+            stripeAccount,
+          },
+        });
+      } else {
+        res.status(500).json({ error: "Unable to connect to Stripe API" });
+      }
+    })
+    .catch((error) => {
+      console.log(error);
```
Comment (sundayezeilo) [inline server/controllers/stripe.js]: You can use `console.error` for this one

**Review by sundayezeilo — CHANGES_REQUESTED** [review]
Hey @RetiFier, good work on this one. Please take al look at those comments and address them.

**Review by moffatethan — APPROVED** [review]

**Pull request merged**